spec/rails_helper.rb

```diff
@@ -8,6 +8,33 @@
 abort('The Rails environment is running in production mode!') if Rails.env.production?
 require 'rspec/rails'
 
+Capybara.register_driver :remote_selenium_headless do |app|
+  options = Selenium::WebDriver::Chrome::Options.new
+  options.add_argument('--headless=new')
+  options.add_argument('--no-sandbox')
+  # options.add_argument('--disable-dev-shm-usage')
+#   options.add_argument('--disable-gpu')
+  options.add_argument('--window-size=2560,1344')
+  options.add_argument('--disable-smooth-scrolling')
+  capabilities = [
+    options,
+    Selenium::WebDriver::Remote::Capabilities.new(
+      'goog:loggingPrefs' => {
+        browser: 'ALL', driver: 'ALL'
+      }
+    )
+  ]
+
+  Capybara::Selenium::Driver.new(app,
+    browser: :remote,
+    capabilities:,
+    url: "http://#{ENV['SELENIUM_HOST'] || 'selenium'}:4444/",
```

Comment:
Try copying exactly what [we got working in TIND-QA](https://github.com/BerkeleyLibrary/tind-qa/blob/main/spec/spec_helper.rb), which is a simplified version of what's configured for Framework. The option arguments and preferences differ a bit between the two and it's not immediately obvious what would cause a problem, so best to start with a known working config.

Reply:
Ah ha! Per our discussion off-GitHub—the problem is that the Selenium container makes requests to localhost:3000, but the actual address (from its perspective) is `app:3000`. IIRC from having worked with Capybara like this in the past, something in the stack (Chrome?) will complain about hitting a hostname with an invalid TLD, so you may have to use `app.test:3000` with a [network alias added to your compose file](https://docs.docker.com/compose/compose-file/compose-file-v3/#networks):

```yaml
services:
  app:
    networks:
      default:
        aliases:
          - app.test
```

Reply:
I also had to reconfigure `shm`:
1. Removed `shm_size` from the Compose file
2. Added the Chrome argument `--disable-dev-shm-usage`

Having done both of those I'm now able to run the feature specs locally on an M-chip MacBook.